xapian-core/weight/pl2weight.cc

```diff
@@ -78,13 +78,52 @@ PL2Weight::init(double)
 
```

Comment:
I've proved that we do indeed want to maximise x in this case, and implemented this in 01f7233a19460ba555f3cc13d964e862250668ca.